internal/fs/fs.go

```diff
@@ -1675,22 +1675,26 @@ func (fs *fileSystem) createLocalFile(
 
 	fullName := inode.NewFileName(parent.Name(), name)
 	child, ok := fs.localFileInodes[fullName]
-	if !ok {
-		var result *inode.Core
-		result, err = parent.CreateLocalChildFile(name)
-		if err != nil {
-			return
-		}
 
-		child = fs.mintInode(*result)
-		fs.localFileInodes[child.Name()] = child
+	if ok && !child.(*inode.FileInode).IsUnlinked() {
+		return
+	}
 
-		// Empty file is created to be able to set attributes on the file.
-		fileInode := child.(*inode.FileInode)
-		err = fileInode.CreateEmptyTempFile()
-		if err != nil {
-			return
-		}
+	// Create a new inode if a local file is unlinked and then recreated with the same name.
```

Comment:
update the comment. it is created for the first time as well.

Reply:
I thought it's an obvious/default scenario that's why I haven't added. Added comment for the change only. 

But feel it is confusing so anyway I added the comment.